### DISCUSSION TITLE
ListBox does not update its layout when put inside a container with RenderTransform transition.

### DISCUSSION THREAD

**Discussion by LilyStilson** [AvaloniaUI/Avalonia#17670] [Q&A] [answered]
I have a custom expander style that has animation on expansion of its content. The animation is done using `<TransformOperationsTransition />` on the parent container of `<ContentPresenter />`. Height is not being changed at any point, I've used `Avalonia.Simple` theme's Expander as a base for this one.

I've recorded a video that demonstrates this behavior. I want to make sure whether it is indented or not and is there a way to fix it?
Without that expansion transition, there's no issue. The video also shows that all other elements work perfectly fine with the transition.

https://github.com/user-attachments/assets/05352219-2ba1-45db-8eeb-4a5aa5ac94ae

The code for the expander style is [here](https://github.com/Night-Sky-Studio/AErender-Launcher/blob/feat%2Ftheme/src%2FAErenderLauncher.Theme%2FControls%2FExpander.axaml)
And the Main Window's ListBox is [here](https://github.com/Night-Sky-Studio/AErender-Launcher/blob/feat%2Ftheme/src%2FAErenderLauncher%2FViews%2FMainWindow.axaml#L130)

**Comment by StefanKoell** [comment]
I've seen a similar issue. If I'm not mistaken the ListBox is using a virtualized stack panel by default. I was able to workaround the issue by switching to a normal stack panel. Would be interesting if this is also the case for you.

**Reply by LilyStilson** [reply]
I don't think you can bind to `<StackPanel />`, you can to `<ItemsRepeater />` though. I've tried the latter option - same issue.

https://github.com/user-attachments/assets/7d18610e-1132-4def-809f-1ee86865e146

**Reply by StefanKoell** [reply, marked as answer]
This is what I did in my ListBox:
```xml
            <ListBox.ItemsPanel>
                <ItemsPanelTemplate>
                    <StackPanel />
                </ItemsPanelTemplate>
            </ListBox.ItemsPanel>
```

Also, what I noticed myself: ItemsRepeater is a weird beast which does weird stuff. When I don't use a ListBox, I usually use an ItemsControl instead.

**Reply by LilyStilson** [reply]
Sorry, I misunderstood what you meant here, I didn't know you could change the underlying panel in ListBox x)
After changing it to `<StackPanel />` the issue went away, thanks :D

**Reply by StefanKoell** [reply]
So it seems that the virtualized panel which is used by default has some issues. Maybe it's worth opening an issue with a small repro project if you want.
